pkg/globalinit/catalog_cache_init.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package globalinit
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"sort"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/catalog"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/lastversion"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/pluginsupplier"
+)
+
+// This global initializer checks if the last executed CLI version is < 1.3.0.
+// If so it refreshes the plugin catalog to make sure any remapping data is in the catalog.
+
+func init() {
+	RegisterInitializer("Plugin Info Catalog Initializer", triggerForPreCommandRemapping, refreshPluginCatalog)
+}
+
+func triggerForPreCommandRemapping() bool {
+	// If the last executed CLI version is < 1.3.0, we need to refresh the plugin catalog.
+	return lastversion.GetLastExecutedCLIVersion() == lastversion.OlderThan1_3_0
+}
+
+// refreshPluginCatalog reads the info from each installed plugin
+// and updates the plugin catalog with the latest info.
+// We need to do this to make sure the command re-mapping data is in the cache.
+func refreshPluginCatalog(outStream io.Writer) error {
+	plugins, err := pluginsupplier.GetInstalledPlugins()
+	if err != nil {
+		return err
+	}
+	sort.Sort(cli.PluginInfoSorter(plugins))
+
+	c, err := catalog.NewContextCatalogUpdater("")
+	if err != nil {
+		return err
+	}
+	defer c.Unlock()
+
+	for i := range plugins {
+		fmt.Fprintf(outStream, "Refreshing plugin %s...\n", plugins[i].Name)
```

Comment:
When printing this line for every plugin, the output gets filled with too much information.
```
$ tz plugin list
Some initialization of the CLI is required.
Let's set things up for you.  This will just take a few seconds.

Refreshing plugin apply...
Refreshing plugin appsv2...
Refreshing plugin build...
Refreshing plugin builder...
Refreshing plugin cluster...
Refreshing plugin clustergroup...
Refreshing plugin ekscluster...
Refreshing plugin iam...
Refreshing plugin package...
Refreshing plugin policy...
Refreshing plugin project...
Refreshing plugin secret...
Refreshing plugin space...
Refreshing plugin telemetry...

Initialization done!
==
```

WDYT if we replace this with a single-line?
```
$ tz plugin list
Some initialization of the CLI is required.
Let's set things up for you.  This will just take a few seconds.

Refreshing installed plugins...

Initialization done!
==
```

Reply:
I think a single line is fine but I’ll also add the extra verbosity lines since you said things hung, that way we can see which plug-in may cause a problem